lib/cylc/rundb.py

```diff
@@ -480,33 +561,224 @@ def select_task_states_by_task_ids(self, keys, task_ids=None):
                 ret[(name, cycle)][key] = value
         return ret
 
-    def select_task_states_by_cycles(self, keys, cycles=None):
-        """Select items from task_states by cycles.
+    def select_task_pool(self, callback, id_key=None):
```

Comment:
They do now, but then again, the pattern is about 2 lines of code in each method, so I am not too annoyed.